src/node_http_parser.cc

```diff
@@ -245,47 +255,48 @@ class Parser : public BaseObject {
     if (!cb->IsFunction())
       return 0;
 
-    Local<Object> message_info = Object::New(env()->isolate());
+    Local<Array> message_info =
+        Array::New(env()->isolate(), kMessageInfoLength);
 
     if (have_flushed_) {
       // Slow case, flush remaining headers.
       Flush();
     } else {
       // Fast case, pass headers and URL to JS land.
-      message_info->Set(env()->headers_string(), CreateHeaders());
+      message_info->Set(kHeaders, CreateHeaders());
       if (parser_.type == HTTP_REQUEST)
-        message_info->Set(env()->url_string(), url_.ToString(env()));
+        message_info->Set(kUrl, url_.ToString(env()));
     }
     num_fields_ = num_values_ = 0;
 
     // METHOD
     if (parser_.type == HTTP_REQUEST) {
-      message_info->Set(env()->method_string(),
-                        Uint32::NewFromUnsigned(env()->isolate(),
-                                                parser_.method));
+      message_info->Set(
+          kMethod, Uint32::NewFromUnsigned(env()->isolate(), parser_.method));
     }
 
     // STATUS
     if (parser_.type == HTTP_RESPONSE) {
-      message_info->Set(env()->status_code_string(),
+      message_info->Set(kStatusCode,
                         Integer::New(env()->isolate(), parser_.status_code));
-      message_info->Set(env()->status_message_string(),
-                        status_message_.ToString(env()));
+      message_info->Set(kStatusMessage, status_message_.ToString(env()));
     }
 
     // VERSION
-    message_info->Set(env()->version_major_string(),
+    message_info->Set(kVersionMajor,
                       Integer::New(env()->isolate(), parser_.http_major));
-    message_info->Set(env()->version_minor_string(),
+    message_info->Set(kVersionMinor,
                       Integer::New(env()->isolate(), parser_.http_minor));
 
-    message_info->Set(env()->should_keep_alive_string(),
-                      http_should_keep_alive(&parser_) ?
-                          True(env()->isolate()) : False(env()->isolate()));
+    message_info->Set(kShouldKeepAlive,
+                      http_should_keep_alive(&parser_)
+                          ? True(env()->isolate())
+                          : False(env()->isolate()));
```

Comment:
While you're here, you could change this to `Boolean::New(env()->isolate(), http_should_keep_alive(&parser_))`.  I originally wrote it like that because there was only the slow Boolean::New() that looks up the isolate in thread-local storage.